src/modules/RootPart.rb

```diff
@@ -2435,6 +2431,28 @@ def mount_regular_fstab_entry?(entry)
       # #MountVarIfRequired... except when /var is a subvolume
       path != "/var" || mntops.include?("subvol=")
     end
+
+    # Creates a pre-update snapshot and stores its number
+    #
+    # If something goes wrong, it reports the problem to the user.
+    def create_pre_snapshot
+      return unless Yast2::FsSnapshot.configured?
+
+      original_scr = WFM.SCRGetDefault()
+      chroot_scr = WFM.SCROpen("chroot=#{Installation.destdir}:scr", false)
+      WFM.SCRSetDefault(chroot_scr)
+      # as of bsc #1092757 snapshot descriptions are not translated
+      snapshot = Yast2::FsSnapshot.create_pre("before update", cleanup: :number, important: true)
+      Yast2::FsSnapshotStore.save("update", snapshot.number)
+    rescue Yast2::SnapshotCreationFailed
+      Yast::Report.Error(
+        _("A pre-update snapshot could not be created. You can continue with the " \
+          "installation, but beware that you cannot roll back to a pre-update state " \
+          "unless you have created a snapshot manually.")
+      )
+    ensure
+      WFM.SCRSetDefault(original_scr) if original_scr
```

Comment:
when I see this maybe it would be nice to have block version in ruby bindings itself for such operation?

Reply:
but probably just for master

Reply:
OK, I will have it into account when merging these changes in master.